recipes/hadoop_yarn_nodemanager.rb

```diff
@@ -52,7 +52,13 @@
 end
 
 # Ensure permissions for secure Hadoop... this *should* be no-op
-file '/usr/lib/hadoop-yarn/bin/container-executor' do
+if node['hadoop']['distribution'] == 'hdp' && node['hadoop']['distribution_version'].to_f == 2.2
+  container_executor_path = '/usr/hdp/current/hadoop-yarn-nodemanager/bin'
+else
+  container_executor_path = '/usr/lib/hadoop-yarn/bin'
+end
```

Comment:
Don't use this syntax. Use the same syntax as every other conditional in the cookbook...
```ruby
variable =
  if condition
    response
  else
    other_response
  end
```